sdk/python/runtime/main.go

```diff
@@ -157,6 +169,16 @@ func (m *PythonSdk) Load(ctx context.Context, modSource *dagger.ModuleSource) (*
 	if err := m.Discovery.Load(ctx, modSource); err != nil {
 		return nil, fmt.Errorf("runtime module load: %w", err)
 	}
+
+	id, err := m.Discovery.ModSource.ID(ctx)
+	if err != nil {
+		return nil, err
+	}
+	h := sha256.New()
+	fmt.Fprint(h, id)
+	idHex := hex.EncodeToString(h.Sum(nil))
+	m.SourcePath = path.Join(ModSourceDirPath, idHex)
```

Comment:
Hm, this isn't the right stable ID to use - it *should* be related to the context directory I think?

For git sources, maybe we could use the commit ID? And for local sources, the `Digest`? Maybe we want a `ModuleSource.digest` property more generally for things like this.

Reply:
> Maybe we want a `ModuleSource.digest` property more generally for things like this.

I've found the need for a unique ID per module to use as a key for a cache volume, for `node_modules` and `.venv`, but was able to avoid it (just not creating a cache volume for it). So I agree that it can be useful in general.

We'll soon have isolated/namespaced cache volumes per module though. Even though that'll apply to the runtime module itself and not the target module, maybe the same approach that @sipsma is going for can be reused here, even if just the part that uniquely identifies a modules. Not sure if it's a simple namespace based on the module's name (like type namespaces) though.

Reply:
Isolated cache volumes for each SDK instance would I think solve this issue, but the problem then is that we wouldn't be able to cache any *legitimately* shared dependencies if they were entirely separated :cry:

I guess the uv cache is just not designed to be shared across multiple hosts, so, we need some little trick just for those paths :cry:

Reply:
Yeah, I know, I was saying maybe you can reuse some part of it, in order to get the unique value for the mount. Unless the idea for the namespaced cache valumes is to just use the module name.

Reply:
*Oh* right I see :tada: :tada: (coffeeee time)

Reply:
I ended up implementing `Directory.digest` and `ModuleSource.digest` so I could have a better way of doing this.

For some reason, the `ID` kept changing, I'm guessing `normalizeCallerLoadedSource` isn't helping here? Having a `Digest` helps keep things a bit more consistent and content-based.